### ISSUE TITLE
prefer-number-properties can't detect usages via global object

### ISSUE THREAD

**Issue by EvgenyOrekhov** [sindresorhus/eslint-plugin-unicorn#1439]
I think `prefer-number-properties` should detect usages via `global`/`window`/`self`:

```js
globalThis.isNaN(foo);
global.isNaN(foo);
window.isNaN(foo);
self.isNaN(foo);

globalThis.isFinite(foo);
global.isFinite(foo);
window.isFinite(foo);
self.isFinite(foo);
```

**Comment by fisker** [comment]
Actually, I tried to use [this](https://eslint-utils.mysticatea.dev/api/scope-utils.html#referencetracker-class) to track global `isNaN` when I implement this rule, the reason I give up this solution is it can't report anything if `{globals: {isNaN: "off"}}` which I already did in my personal config. But maybe I was wrong? Welcome to give a shot.

**Comment by EvgenyOrekhov** [comment]
Also added `globalThis` to examples.
